I18n(ja): Update translation of deploy.md (#1970)
withastro/docs: src/pages/ja/guides/deploy.md

---
title: Astroサイトのデプロイ
description: WebにAstroサイトをデプロイする方法。
layout: ~/layouts/MainLayout.astro
setup: import DeployGuidesNav from '~/components/DeployGuidesNav.astro';
---
**Astroサイトをビルド・デプロイする準備ができましたか？** さまざまなデプロイサービスのガイドに従うか、スクロールしてAstroサイトをデプロイする全般的なガイダンスを参照してください。

<DeployGuidesNav />

## クイックデプロイ・オプション

> いくつかのホストに対しては、それらのWebサイトのダッシュボードUIまたはCLIのいずれかを使って、Astroサイトをビルド・デプロイできます。

Reviewer: ここちょっと日本語がわかりづらいように感じました。たとえば、こんな感じはどうでしょう？

```suggestion
Astroのサイトは、WebサイトのダッシュボードUIまたはCLIを使用して、多くのホストに短時間でビルド・デプロイできます。
```

### WebサイトUI

> Webサイトを素早くデプロイする方法は、AstroプロジェクトのオンラインGitリポジトリ（例：GitHub, GitLab, Bigbucket）をホストプロバイダーに接続し、Gitを使用した継続的デプロイを使用することです。

Reviewer: A quick way to deploy なので、素早くデプロイというよりは手っ取り早い方法とかのほうがいいかなと思いました。
あとBitbucketがBigbucketになってました。

```suggestion
Webサイトをデプロイする手っ取り早い方法は、AstroプロジェクトのオンラインGitリポジトリ（例：GitHub, GitLab, Bitbucket）をホストプロバイダーに接続し、Gitを使用した継続的デプロイを使用することです。
```

これらのホストプラットフォームは自動的にAstroプロジェクトのソースリポジトリへのプッシュを検出し、サイトをビルドして、カスタムURLまたは個人のドメインでWebにデプロイします。多くの場合、これらのプラットフォームにおけるデプロイの設定は以下のような手順に従います：

1. リポジトリをオンライのGitプロバイダー（例：GitHub, GtiLab, BitBucket）に追加してください。

1. **継続的デプロイ**をサポートするホスト（例：[Netlify](/ja/guides/deploy/netlify/) または [Vercel](/ja/guides/deploy/vercel/)）を選択し、新しいサイト/プロジェクトとしてGitリポジトリをインポートしてください。

>   多くの一般的なホストはプロジェクトをAstroサイトとして認識し、以下のようにビルド・デプロイに適切な構成設定を選択するはずです。（もしそうでない場合は、これらの設定は変更できます。）

Reviewer 2: I can confirm @kyosuke’s suspicion. This needs to be indented two spaces more:

```suggestion
    多くの一般的なホストはプロジェクトをAstroサイトとして認識し、以下のようにビルド・デプロイに適切な構成設定を選択するはずです。（もしそうでない場合は、これらの設定は変更できます。）
```

>   :::note[デプロイの設定]
>   - **ビルドコマンド:** `astro build` または `npm run build`
>   - **公開ディレクトリ:** `dist`
>   :::

Reviewer 2: And same here:

```suggestion
    :::note[デプロイの設定]
    - **ビルドコマンド:** `astro build` または `npm run build`
    - **公開ディレクトリ:** `dist`
    :::
```

>   :::

Reviewer: ここ、インデントが合っていないためでしょうか。英語版と比べるとインデント位置がずれているようなのでご確認いただけますか？

Author: 失礼しました！確認します。

> 1. 「デプロイ」をクリックすると、新しいWebサイトがそのホストのユニークなURLに作成されます（例： `new-astro-site.netlify.app`）。

Reviewer: 多分上のリストの一部となっていないため、3.ではなく、1.と表示されてしまっています。

ホストはGitプロバイダーのmainブランチの変更を監視し、新しいコミットがあるたびにサイトを再ビルド・再公開するよう自動的に設定します。これらの設定は通常、ホストプロバイダーのダッシュボードUI上で行うことができます。

### CLIを使用したデプロイ

いくつかのホストは、npmを使用したあなたのマシンにグローバルにインストールできる独自のコマンドライン・インターフェース（CLI）を備えています。多くの場合、CLIを使用してデプロイする方法は以下のようになります:

1. ホストのCLIをグローバルにインストールします。例：

    ```bash
    npm install --global netlify-cli
    ```

1. CLIを実行し、認証、セットアップ等の指示に従います。

1. サイトをビルドしてホストにデプロイします。

   多くの一般的なホストは、サイトをビルドしデプロイしてくれます。それらは通常、プロジェクトをAstroサイトとして認識し、以下のようにビルド・デプロイのための適切な設定を選択するはずです。（もしそうでない場合は、これらの設定は変更可能です。）

    :::note[デプロイの設定]
    - **ビルドコマンド:** `astro build` or `npm run build`
    - **公開ディレクトリ:** `dist`
    :::

    その他のホストでは[ローカルでサイトをビルド](#ローカルでサイトをビルドする)しコマンドラインを使用してデプロイする必要があります。

## ローカルでサイトをビルドする

NetlifyやVercelのような多くのホストはサイトをビルドしてそのビルド出力をWebに公開してくれます。しかし、サイトによっては、ローカルでビルドし、デプロイコマンドを実行するか、ビルド出力をアップロードする必要があります。

また、ローカルでビルドしてサイトをプレビューしたり、潜在的なエラーや警告を自分の環境でキャッチしたりすることもできます。

Astroサイトをビルドするには、`npm run build`コマンドを実行してください。

```bash
npm run build
```

デフォルトでは、ビルド出力は`dist/`に配置されます。この場所は[`outDir`設定オプション](/ja/reference/configuration-reference/#outdir)を使って変更できます。

## SSR用アダプターを追加する

:::note

> [SSR（サーバーサイド・レンダリング）](/ja/guides/server-side-rendering/)を有効化してサイトをデプロイする前に、確認してください:

Reviewer: ここのAstroは省略しても意味的に問題ないと思いますが、原文にあったので一応。

```suggestion
[SSR（サーバーサイド・レンダリング）](/ja/guides/server-side-rendering/)を有効化してAstroサイトをデプロイする前に、確認してください:
```

> - [適切なアダプター](/ja/guides/server-side-rendering/#enabling-ssr-in-your-project)がプロジェクトの依存関係にインストールされていること（手動、または、アダプターの`astro add`コマンドを使用して。例: `npx astro add netlify`）。

Reviewer: 使用して。で終わってるのが気になりました。

```suggestion
- [適切なアダプター](/ja/guides/server-side-rendering/#enabling-ssr-in-your-project)がプロジェクトの依存関係にインストールされていること（手動、またはアダプターの`astro add`コマンドを使用します。例: `npx astro add netlify`）
```

> - 手動インストール時に`astro.config.mjs`ファイルのimportとdefault exportに [アダプター](/ja/reference/configuration-reference/#インテグレーション) を追加した。(このステップは `astro add` コマンドが代行してくれます！)

Reviewer: 一つ上の項目と語尾を合わせたほうがよさそうです。

```suggestion
- 手動インストール時に`astro.config.mjs`ファイルのimportとdefault exportに[アダプター](/ja/reference/configuration-reference/#インテグレーション)を追加していること（このステップは `astro add` コマンドが代行します！）
```
